specs/src/tx_client.md

```diff
@@ -0,0 +1,253 @@
+
+## Submission
+
+**Important**: Currently, only the below submission patterns are reliable for submitting transactions. Other patterns like sending multiple transactions from one account without waiting for confirmation will likely cause sequence mismatches and failures.
+
+### Ordered Submission
+
+**Characteristics**: Low Throughput, Sequential Submission, Better ordering guarantees
+
+- **Single Account**: All transactions signed by one account
+- **Sequential**: Max one tx per block, each tx must be confirmed before subsequent can be resubmitted
+- **Ordering Guarantees**: Transactions are processed in exact submission order
+- **Low Throughput**: One tx per block per account
+- **Use Case**:
+  - Applications requiring transaction ordering
+  - Users who require that all their blobs are signed by the same account.
+
+**APIs**:
+
+- **SubmitTx**: Sequential submission for SDK txs (no worker queue).
+  - **Process**: Build transaction → Broadcast → Confirm
+
+    ```go
+    func (client *TxClient) SubmitTx(ctx context.Context, msgs []sdktypes.Msg, opts ...TxOption) (*TxResponse, error)
+    ```
+
+- **SubmitPayForBlob**: Sequential submission with default account (no worker queue).
+  - **Process**: Create PFB with default account → Broadcast → Confirm
+
+    ```go
+    func (client *TxClient) SubmitPayForBlob(ctx context.Context, blobs []*share.Blob, opts ...TxOption) (*TxResponse, error)
+    ```
+
+- **SubmitPayForBlobWithAccount**: Sequential submission with a specified account (bypasses worker queue).
+  - **Process**: Create PFB with specified account → Broadcast → Confirmation
+
+    ```go
+    func (client *TxClient) SubmitPayForBlobWithAccount(ctx context.Context, accountName string, blobs []*share.Blob, opts ...TxOption) (*TxResponse, error)
+    ```
+
+### Unordered Submission
+
+**Characteristics**: High Throughput, Parallel submission, No Ordering
+
+- **Worker Accounts**: A pool of accounts is created by the primary account, each fee-granted with unlimited allowances.
+- **Worker Queue**: Transactions are queued and dispatched to available workers.
+- **Parallel Workers**:
+  - Worker 0 = primary account
+  - Workers >0 = additional fee-granted accounts
+- **High Throughput**: Multiple transactions can be submitted and processed per block concurrently, without sequence contention.
+- **No Ordering Guarantees**: Execution order is nondeterministic; workers process transactions as they become available.
+- **Fee Grants**: All fees are covered by the primary account, so workers operate without individual balance management.
+- **Account Reuse**: Worker accounts are persisted across runtimes when parallel submission is enabled.
+- **Use Case**: Best suited for applications that prioritize throughput over ordering.
+
+**note:** If initialized with only one worker, this mode behaves identically to Sequential Submission (one tx per block).
+
+**APIs**:
```

Comment:
is the `APIs` part of the spec bonding? If so, then I'd say it should be less specific, on rust side having things like `resultC` would feel weird